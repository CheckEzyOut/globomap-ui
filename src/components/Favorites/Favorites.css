/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

.favorites {
  position: fixed;
  top: 0;
  left: 100px;
  bottom: 0;
  right: 0;
  padding: 50px 7% 0;
  z-index: 8;
  background: white;
}

.favorites-title {
  font-size: 2em;
  height: 42px;
  margin: 0 0 30px;
}

.user-map-list {
  list-style: none;
  padding: 0;
}

.user-map-item {
  font-size: 18px;
  border-bottom: 1px solid #eee;
  padding: 15px 5px;
}

.user-map-item:hover {
  cursor: pointer;
  color: #2eb78e;
}

.user-map-item:last-child {
  border-bottom: none;
}

.user-map-item-remove-btn {
  float: right;
  border: 0;
  background: none;
  color: #999;
}

.user-map-item:hover .user-map-item-remove-btn {
  color: #ff3860;
}
