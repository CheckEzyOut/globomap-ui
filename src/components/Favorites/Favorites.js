/*
Copyright 2018 Globo.com

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

import React from 'react';
import { connect } from 'react-redux';
import {
  listUserMaps,
  deleteUserMap,
  getUserMap,
  setStageNodes
} from '../../redux/modules/stage';
import {
  clearCurrentNode,
  resetSubNodes
} from '../../redux/modules/nodes';
import { setTab } from '../../redux/modules/tabs';
import { sortByName } from '../../utils';
import './Favorites.css';

class Favorites extends React.Component {

  componentWillMount() {
    this.props.listUserMaps();
  }

  applyGraph(event, content) {
    event.stopPropagation();
    this.props.clearCurrentNode();
    this.props.resetSubNodes();
    this.props.setStageNodes(content);
    this.props.setTab('map');
  }

  onDeleteGraph(event, key) {
    event.stopPropagation();
    if (window.confirm('Are you sure to delete this item?')) {
      this.props.deleteUserMap(key);
    }
  }

  renderUserMaps() {
    if (this.props.userMaps.length === 0) {
      return (
        <ul className="user-map-list">
          <li className="user-map-item no-content">No map saved yet</li>
        </ul>
      );
    }

    const uMaps = sortByName(this.props.userMaps);
    return (
      <ul className="user-map-list">
        {uMaps.map(item => {
          return (
            <li key={item.key} className="user-map-item"
                onClick={e => this.applyGraph(e, item.content)}>
              <span  className="user-map-item-title">{item.name}</span>
              <button className="user-map-item-remove-btn"
                      onClick={e => this.onDeleteGraph(e, item.key)}
                      title="Delete this item">
                <i className="fa fa-times"></i>
              </button>
            </li>
          );
        })}
      </ul>
    );
  }

  render() {

    return (
      <div className="favorites">
        <h3 className="favorites-title">Saved maps</h3>
        {this.renderUserMaps()}
      </div>
    );
  }

}

function mapStateToProps(state) {
  return {
    hasId: state.app.hasId,
    userMaps: state.stage.userMaps
  };
}

export default connect(
  mapStateToProps,
  {
    clearCurrentNode,
    listUserMaps,
    deleteUserMap,
    getUserMap,
    resetSubNodes,
    setStageNodes,
    setTab
  }
)(Favorites);

