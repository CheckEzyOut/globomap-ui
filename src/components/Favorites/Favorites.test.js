import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import { Favorites } from './Favorites';

describe("Favorites", function() {

  let componentWillMountStub;

  beforeEach(function() {
    componentWillMountStub = sinon.stub(Favorites.prototype, 'componentWillMount').value('newValue');
  });

  afterEach(function () {
    componentWillMountStub.restore();
  });

  it('should render Favorites component', () => {
    const props = {
      userMaps: []
    };
    const component = shallow(<Favorites {...props} />);
    expect(component).toMatchSnapshot();
  });

});
