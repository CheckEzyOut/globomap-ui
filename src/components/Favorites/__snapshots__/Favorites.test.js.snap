// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should render Favorites component 1`] = `
<div
  className="favorites"
>
  <h3
    className="favorites-title"
  >
    Saved maps
  </h3>
  <ul
    className="user-map-list"
  >
    <li
      className="user-map-item no-content"
    >
      No map saved yet
    </li>
  </ul>
</div>
`;
